src/modules/didRegistry/didRegistry.service.ts

```diff
@@ -0,0 +1,271 @@
+    private _did: string;
+    private _document: IDIDDocumentFull;
+    private _ipfsStore: DidStore;
+    private _jwt: JWT;
+    private _userClaims: ClaimsUser;
+    private _issuerClaims: ClaimsIssuer;
+
+    constructor(
+        private _signerService: SignerService,
+        private _cacheClient: CacheClient,
+        private _assetsService: AssetsService,
+        private _ipfsUrl = "https://ipfs.infura.io:5001/api/v0/",
+    ) {
+        this._signerService.onInit(this.init.bind(this));
+    }
+
+    static async connect(
+        signerService: SignerService,
+        cacheClient: CacheClient,
+        assetsService: AssetsService,
+        ipfsUrl?: string,
+    ) {
+        const registry = new DidRegistry(signerService, cacheClient, assetsService, ipfsUrl);
+        await registry.init();
+        return registry;
+    }
+
+    get jwt() {
+        return this._jwt;
+    }
+
+    // temporarily, to allow claim service to save claim
+    get ipfsStore() {
+        return this._ipfsStore;
+    }
+
+    async init() {
+        this._ipfsStore = new DidStore(this._ipfsUrl);
+        await this._setOperator();
+        this.setJWT();
+        await this._setDocument();
+        this._setClaims();
+    }
+
+    async getDidDocument({
+        did = this._did,
+        includeClaims = true,
+    }: { did?: string; includeClaims?: boolean } | undefined = {}) {
+        if (this._cacheClient) {
+            try {
+                const didDoc = await this._cacheClient.getDidDocument(did, includeClaims);
+                return {
+                    ...didDoc,
+                    service: didDoc.service as (IServiceEndpoint & ClaimData)[],
+                };
+            } catch (err) {
+                if ((err as AxiosError).response?.status === 401) {
+                    throw err;
+                }
+                console.log(err);
+            }
+        }
+
+        const document = await this._operator.read(did);
+        return {
+            ...document,
+            service: includeClaims
+                ? await this.downloadClaims({
+                      services: document.service && document.service.length > 0 ? document.service : [],
+                  })
+                : [],
+        };
+    }
+
+    /**
+     * createPublicClaim
+     *
+     * @description create a public claim based on data provided
+     * @returns JWT token of created claim
+     *
+     */
+    async createPublicClaim({ data, subject }: { data: ClaimData; subject?: string }) {
+        if (subject) {
+            return this._userClaims.createPublicClaim(data, { subject, issuer: "" });
+        }
+        return this._userClaims.createPublicClaim(data);
+    }
+
+    async createProofClaim({
+        claimUrl,
+        encryptedSaltedFields,
+    }: {
+        claimUrl: string;
+        encryptedSaltedFields: IProofData;
+    }) {
+        await this._userClaims.createProofClaim(claimUrl, encryptedSaltedFields);
+    }
```

Comment:
Should these `claim` methods be in the `didRegistry` module?

Reply:
Logically they belong to claims module. I've placed them here because having them in claims module will just duplicate code because claims module doesn't add anything to creation. Also may be `createProofClaim` can be deleted as it is not used anywhere. Also to justify they are presence here we can see them as tokens, which are part of actual claim

Reply:
At the moment, with your current PR, they are also in the `claims` modules. So I think they are already duplicated?
I don't understand why we can't remove them from `didRegistry` module?

I don't understand your point about "seeing them as tokens". What are we seeing as tokens?

Re removing `createProofClaim`, I agree, let's remove it 👍

Reply:
Indeed. It is already in claims service. I `createPublicClaim` is put in claims module and not duplicated in did module, then userClaims will be exported from claims, which seems as internal implementation of did module.
By differentiating between token and claim I mean that value returned from `createPublicClaim` is a part of type `Claim`

Reply:
Ok, thanks, I'm understanding better now, I think it is okay for now 👍